Autopilot/PathManager/Src/landingManager.cpp

```diff
@@ -0,0 +1,200 @@
+
+    iPoint[0] = intersectionPoint->longitude;
+    iPoint[1] = intersectionPoint->latitude;
+    iPoint[2] = intersectionPoint->altitude;
+
+    sPoint[0] = stoppingPoint->longitude;
+    sPoint[1] = stoppingPoint->latitude;
+    sPoint[2] = stoppingPoint->altitude;
+
+    cPoint[0] = input->longitude;
+    cPoint[1] = input->latitude;
+    cPoint[2] = input->altitude;
+
+    //calculating the vectors
+    float vectorAI[3];
+    float vectorAS[3];
+    float vectorAC[3];
+    float normal[3];
+
+    //aiming point to intersection point
+    vectorAI[0] = iPoint[0] - aPoint[0];
+    vectorAI[1] = iPoint[1] - aPoint[1];
+    vectorAI[2] = iPoint[2] - aPoint[2];
+
+    //aiming point to stopping point
+    vectorAS[0] = sPoint[0] - aPoint[0];
+    vectorAS[1] = sPoint[1] - aPoint[1];
+    vectorAS[2] = sPoint[2] - aPoint[2];
+
+    //aiming point to current point
+    vectorAC[0] = cPoint[0] - aPoint[0];
+    vectorAC[1] = cPoint[1] - aPoint[1];
+    vectorAC[2] = cPoint[2] - aPoint[2];
+    
+    //normal of the plane created by aiming, stopping, and intersection point
+    normal[0] = vectorAI[1]*vectorAS[2] - vectorAI[2]*vectorAS[1];
+    normal[1] = vectorAI[2]*vectorAS[0] - vectorAI[0]*vectorAS[2]; //flipped negative
+    normal[2] = vectorAI[0]*vectorAS[1] - vectorAI[1]*vectorAS[0];
+
+    //projecting vectorAC to the normal
+    //equation is the dot product of normal and vectorAC, divided by the norm of the normal squared, multiplied by the normal vector
+    float projectionACToNormal[3];
+
+    projectionACToNormal[0] = (normal[0]*vectorAC[0] + normal[1]*vectorAC[1] + normal[2]*vectorAC[2]) / (pow(normal[0],2) + pow(normal[1],2)+ pow(normal[2],2)) * normal[0];
+    projectionACToNormal[1] = (normal[0]*vectorAC[0] + normal[1]*vectorAC[1] + normal[2]*vectorAC[2]) / (pow(normal[0],2) + pow(normal[1],2)+ pow(normal[2],2)) * normal[1];
+    projectionACToNormal[2] = (normal[0]*vectorAC[0] + normal[1]*vectorAC[1] + normal[2]*vectorAC[2]) / (pow(normal[0],2) + pow(normal[1],2)+ pow(normal[2],2)) * normal[2];
+
+    //project point onto plane (cpoint minus the projection)
+    projectedPoint[0] = cPoint[0] - projectionACToNormal[0];
+    projectedPoint[1] = cPoint[1] - projectionACToNormal[1];
+    projectedPoint[2] = cPoint[2] - projectionACToNormal[2];
+
+    //determine the equation of the line of the slope and sub in projected values (symmetric form)
+    double altitude = (((projectedPoint[0] - aPoint[0]) / vectorAS[0]) + aPoint[2]) * vectorAS[2];
+
+    return altitude;
+}
+/*
+double LandingManager::horizontalAdjustment(_WaypointManager_Data_In input, _PathData * aimingPoint, _PathData * intersectionPoint)
+{
+    //determine path and location
+    //using follow waypoints algorithm, output the desired horizontal track
+}
+*/
+double LandingManager::throttleOff(void)
+{
+    //return an altitude so that the throttle turns off
+    //I'm not sure if this is a good way to control throttle within the path manager
+    //Any suggestions?
```

Comment:
I don't think turning the throttle on and off is the best way to go about this - it means we'll constantly overshoot and the plane will dolphin on the glide path

Reply:
This is only used for flare and decrab, not any other states. Other states will use the control altitude function, which will send a desired altitude to the attitude manager. 

This function is only used to turn the throttle off when we need to during flare and decrab.